packages/core/test-support/helpers/contains-helpers.js

```diff
@@ -27,3 +27,62 @@ export function buttonContains(text) {
 export function linkContains(text) {
   return findByContains('a', text);
 }
+
+/**
+ * This function allows us to keep using the jquery notation for :contains without using jquery
+ *
+ * Example:
+ * Jquery: $(.class-thing:contains(blah) .metric .test:contains(other))
+ *
+ * findByContains:
+ * [...findByContains('.class-thing', 'blah').querySelectorAll('.metric .test')].find(el => el.innerText.trim().includes('other'))
+ *
+ * Non-Jquery Equivalent: [
+ *    ...[
+ *      ...document.querySelectorAll(
+ *        (.class-thing)
+ *      )].find(el => el.innerText.trim().includes('blah'))
+ *    .querySelectorAll('.metric .test')
+ *  ]
+ *  .find(el => el.innerText.trim().includes('other'))
+ *
+ * findContains: findContains(.class-thing:contains(blah) .metric .test:contains(other))
+ *
+ * @param {String} selector - jquery or native style selector
+ * @param {HTMLElement} [baseElement] - optional parameter to use as base element to select off of
+ * @returns {HTMLElement} - Element based on text it (or its parent) contains
+ */
+export function findContains(selector, baseElement) {
+  let nativeSelector,
+    parentElement = baseElement || document;
+
+  if (selector.includes(':contains(')) {
+    let splits = selector.split(':contains(');
+    let firstSelector = splits.shift(); //Selector without the :contains operator
+    let restOfSelector = splits.join(':contains('); //We only want to split on the first instance of :contains(, so join the rest back together if there were more
+    let endOfContentIndex = findEndOfContentIndex(restOfSelector);
+    let content = restOfSelector.substring(0, endOfContentIndex);
+    let nextSelector = restOfSelector.substring(endOfContentIndex + 1).trim();
+    let filteredElement = [...parentElement.querySelectorAll(firstSelector)].find(el =>
+      el.innerText.trim().includes(content)
+    );
```

Comment:
I've provided a regex that handles the above case by taking `.someclass:contains(sometext).otherclass` -> firstSelector = `.someclass.otherclass` and content = `sometext`

Reply:
That's a different selector though, isn't it? That would be equivalent to `.someclass .otherclass:contains(sometext)` instead of `.someclass:contains(sometext) .otherclass`

Reply:
those are equivalent selectors.

Reply:
They are not equivalent. 
<img width="583" alt="Screen Shot 2019-04-01 at 1 29 21 PM" src="https://user-images.githubusercontent.com/23023478/55350687-431d8300-5482-11e9-9a8e-26b89c4c4fcd.png">

Reply:
you put an extra descendants operator, but you are correct.